src/H5Cpkg.h

```diff
@@ -82,424 +75,167 @@
-           )                                                                 \
-       )                                                                     \
-     )                                                                       \
+#define H5C__GEN_DLL_PRE_REMOVE_SC(entry_ptr, list_next, list_prev, head_ptr, tail_ptr, len, list_size, fail_val) \
+if (((head_ptr) == NULL) || ((tail_ptr) == NULL) ||                          \
+    ((entry_ptr) == NULL) || ((len) <= 0) ||                                 \
+    ((list_size) < (entry_ptr)->size) ||                                     \
+    ((entry_ptr)->list_prev == NULL && (head_ptr) != (entry_ptr)) ||        \
+    ((entry_ptr)->list_next == NULL && (tail_ptr) != (entry_ptr)) ||        \
+    ((len) == 1 &&                                                           \
+     !((head_ptr) == (entry_ptr) && (tail_ptr) == (entry_ptr) &&             \
+       (entry_ptr)->list_next == NULL && (entry_ptr)->list_prev == NULL && \
+       (list_size) == (entry_ptr)->size                                      \
+      )                                                                      \
+    )                                                                        \
    ) {                                                                       \
-    HGOTO_ERROR(H5E_CACHE, H5E_SYSTEM, (fail_val), "DLL pre remove SC failed")     \
+    HGOTO_ERROR(H5E_CACHE, H5E_SYSTEM, (fail_val), "DLL pre remove SC failed")               \
 }
 
-#define H5C__DLL_SC(head_ptr, tail_ptr, len, list_size, fail_val)                   \
-if ( ( ( ( (head_ptr) == NULL ) || ( (tail_ptr) == NULL ) ) &&           \
-       ( (head_ptr) != (tail_ptr) )                                      \
-     ) ||                                                                \
-     ( (len) < 0 ) ||                                                    \
-     ( (list_size) < 0 ) ||                                                   \
-     ( ( (len) == 1 ) &&                                                 \
-       ( ( (head_ptr) != (tail_ptr) ) ||                                 \
-         ( (head_ptr) == NULL ) || ( (head_ptr)->size != (list_size) )        \
-       )                                                                 \
-     ) ||                                                                \
-     ( ( (len) >= 1 ) &&                                                 \
-       ( ( (head_ptr) == NULL ) || ( (head_ptr)->prev != NULL ) ||       \
-         ( (tail_ptr) == NULL ) || ( (tail_ptr)->next != NULL )          \
-       )                                                                 \
-     )                                                                   \
-   ) {                                                                   \
-    HGOTO_ERROR(H5E_CACHE, H5E_SYSTEM, (fail_val), "DLL sanity check failed")  \
+#define H5C__GEN_DLL_SC(list_next, list_prev, head_ptr, tail_ptr, len, list_size, fail_val) \
```

Comment:
This macro doesn't seem to be used for anything

Reply:
Ah, interesting...    I looked in older versions of the file and it doesn't appear used in the ones I've looked at either.   I'll remove it.